src/middlewared/middlewared/plugins/etc.py

```diff
@@ -177,12 +184,15 @@ def __init__(self, *args, **kwargs):
             if renderer is None:
                 raise ValueError(f'Unknown type: {entry["type"]}')
 
+            self.args = args
```

Comment:
Setting self.args is not a good idea IMO. It is prone to a race condition.

We should figure out another way to handle this, args sounds like a dirty solution.